meta/Ubuntu18.04LTS/src/DistroLauncher-Appx/MyDistro.appxmanifest

```diff
@@ -0,0 +1,43 @@
+﻿<?xml version="1.0" encoding="utf-8"?>

```

Comment:
Why do we need an override there? It’s better to keep versioned distro with the root version (`[src/DistroLauncher-Appx/MyDistro.appxmanifest`. Let’s really have overrides when we need them.

Reply:
My reasoning was: newer releases will behave as Ubuntu22.04 and Preview, which handle the new command line option to be invoked by the App Installer or Start Menu, but `18.04` and `20.04` don't have to, and because of that they would need their appxmanifest without the `uap10:Parameters` option. So I kept the previous version of the appxmanifest in the `meta/src` for those two releases while kept the root-tree manifest with the new behavior exactly to prevent the need to copy files as new releases appear.

---

I realized now (and just experimented with the idea) that we can avoid touching the build system if we let the code just ignore that option for the older releases. I can take advantage of the empty `extended_messages.h` files which exist to avoid leaking the new command line parsing behavior into the old releases to `#define` a preprocessor constant, let's say `UNSUPPORTED_EXTENDED_CLI`, and based on that constant disregard this and other extended command line options to preserve the original experience for the old releases. With this we won't need any manifest overrides anywhere.

I'm not a big fan of preprocessor macros, but this is a very simple and straightforward way to achieve what I'm looking for. I'm really tempted into pushing this idea forward.

How do you feel about that?

Reply:
Yes, if we were stable in terms of appxmanifest content, I wouldn’t have any issues in freezing it time older releases… So, I think your preprocessor approach would be better (and simply disable options we don’t want to work on those plateforms)

Reply:
Great, thanks! I'll revert the manifests override and do the preprocessor way.